tgg/research/gnomad_tgps.py

```diff
@@ -0,0 +1,354 @@
+Will repartition these temporary files on read.
+"""
+
+PANGENOME_ID_DELIM = " "
+"""
+The delimiter used in the provided text file containing pangenome IDs.
+"""
+
+PAGENOME_ID_COL_NUM = 1
+"""
+The column number of the sample IDs in the provided text file containing pangenome IDs.
+"""
+
+HGDP_SUBSET_MT = (
+    "gs://gcp-public-data--gnomad/release/3.1.2/mt/genomes/"
+    "gnomad.genomes.v3.1.2.hgdp_1kg_subset_sparse.mt"
+)
+"""
+Public Hail matrix table for the HGDP+1KG subset data.
+"""
+
+OUTPUT_FILENAME = "pangenome_assessment_stats.txt"
+"""
+The filename of the output file, which contains the computed statistics for the 
+number of common variants per gnomAD sample.
+"""
+
+PANGENOME_CHECKPOINT_FILEPATH = "gs://gnomad-tmp-4day/pangenome/pangenome_checkpoint.ht"
+"""
+The Google Cloud path to the pangenome Hail table, 
+which contains common SNVs that are present in the pangenome. 
+"""
+
+NOT_IN_PANGENOME_CHECKPOINT_FILEPATH = (
+    "gs://gnomad-tmp-4day/pangenome/not_in_pangenome_checkpoint.ht"
+)
+"""
+The Google Cloud path to the gnomAD matrix table, 
+which contains common variants excluding SNVs present in the pangenome. 
+"""
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("Pangenome_assessment")
+logger.setLevel(logging.INFO)
+
+
+def get_pangenome_ids(
+    id_path: str,
+) -> hl.expr.SetExpression:
+    """
+    Import Hail set of pangenome sample IDs from specified file path.
+
+    :param id_path: Path to text file (stored in Google bucket) containing
+    pangenome sample IDs.
+    :return: SetExpression of pangenome sample IDs.
+    """
+    ids = []
+    with hfs.open(id_path, "r") as file:
+        for line in file:
+            sid = line.split(PANGENOME_ID_DELIM)[PAGENOME_ID_COL_NUM].strip()
+            ids.append(sid)
+    return hl.set(ids)
+
+
+def write_output(
+    var_stats: hl.expr.StructExpression,
+    file_dir: str,
+    filename: str,
+) -> None:
+    """
+    Write the output Struct containing computed stats to a .txt file.
+
+    :param var_stats: Data containing computed variant stats.
+    :param file_dir: Path to a Google bucket for output file.
+    :param filename: Output filename.
+    :return: None; function writes output to the output path.
+    """
+    with hfs.open(f"{file_dir}/{filename}", "w") as f:
+        print(var_stats, file=f)
+
+
+def compute_stats(
+    pangenome_ids_path: str,
+    output_dir: str,
+    overwrite: bool,
+) -> None:
+    """
+    Count the number of common (AF > 1%) variants per gnomAD sample.
+
+    Function produces counts only for gnomAD samples that are not part of the pangenome.
+
+    :param pangenome_ids_path: Path to text file (stored in Google bucket)
+    containing pangenome sample IDs.
+    :param output_dir: Path to a Google bucket for output file.
+    :param overwrite: A Boolean variable to specify whether or not to overwrite an
+    existing file.
```

Comment:
```suggestion
    :param overwrite: A Boolean variable to specify whether or not to overwrite an
    existing file. If True, function will overwrite `PANGENOME_CHECKPOINT_FILEPATH`
        and `NOT_IN_PANGENOME_CHECKPOINT_FILEPATH`. If False, function will attempt to read
        from existing file paths.
```
(my formatting might be off)